client/domain/CorrectnessTestObjectFactory.js

```diff
@@ -22,15 +22,21 @@ tie.factory('CorrectnessTestObjectFactory', [
   function() {
     var CorrectnessTest = function(correctnessTestDict) {
       this._input = correctnessTestDict.input;
+      this._stringifiedInput = JSON.stringify(this._input);
       this._allowedOutputs = correctnessTestDict.allowedOutputs;
       this._message = correctnessTestDict.message;
+      this._tag = correctnessTestDict.tag;
     };
 
     // Instance methods.
     CorrectnessTest.prototype.getInput = function() {
       return this._input;
     };
 
+    CorrectnessTest.prototype.getStringifiedInput = function() {
+      return JSON.stringify(this._input);
```

Comment:
If you are storing the stringified input, and if the input doesn't change, why are you redoing the serialization?

In other words, either do this, or keep line 25, but not both.